### DISCUSSION TITLE
Auto reload

### DISCUSSION THREAD

**Discussion by mano1233** [strimzi/kafka-kubernetes-config-provider#37] [Q&A] [answered]
Hey, is the provider reloading the configuration once there is a change in the K8s secret? I appreciate any help you can provide.

**Comment by scholzj** [comment, marked as answer]
The provider itself does not cache any values. But that does not mean it will somehow update the configs when you update a Secret - the config provider is called by Kafka and only then provides the current value from the Secret / Config Map. When that happens depends on where you use it. It might be when a connector configuration is updated, Connect node is restarted etc.

**Reply by mano1233** [reply]
thanks for the reply!

**Reply by OneCricketeer** [reply]
It's written in the k8s docs itself that pods don't restart when secrets are modified, so I think the answer lies outside of this plugin & "config update" would not see the new value, only a `rollout restart` would finally get the updated value. Just like how you can't externally modify an env-var without a pod restart.

**Reply by scholzj** [reply]
The Kubernetes Config Provider does not read the data from the Secret or Config Map mounted into the Pod. It uses the Kubernetes API to actually get the Secret or Config Map from Kubernetes itself.

So strictly speaking, it does not need a Pod restart to get the new value. But it still needs to be triggered by Kafka to provide the value. That happens, among other things, when the pod is restarted. But it happen in other situations as well. For example, with Connectors in Kafka Connect, it also happens when the connector configuration is updated through the REST API etc.
